extension/observer/cfgardenobserver/config.go

```diff
@@ -4,15 +4,116 @@
 package cfgardenobserver // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/cfgardenobserver"
 
 import (
+	"errors"
+	"fmt"
 	"time"
 )
 
 // Config defines configuration for CF Garden observer.
 type Config struct {
-	// The URL of the CF Garden api. Default is "unix:///var/vcap/data/garden/garden.sock"
-	Endpoint string `mapstructure:"endpoint"`
+	// CloudFoundry API Configuration
+	CloudFoundry CfConfig `mapstructure:"cloud_foundry"`
+
+	// Garden API Configuration
+	Garden GardenConfig `mapstructure:"garden"`
 
 	// RefreshInterval determines the frequency at which the observer
 	// needs to poll for collecting information about new processes.
+	// Default: "1m"
 	RefreshInterval time.Duration `mapstructure:"refresh_interval"`
+
+	// The time to wait before resyncing app information on cached containers
+	// using the CloudFoundry API. Example: cache_sync_interval: "20m"
```

Comment:
```suggestion
	// using the CloudFoundry API.
```
Nit: Config examples are in the README, it's not necessary to include here as well.